python/tvm/relay/function.py

```diff
@@ -63,3 +63,10 @@ def __call__(self, *args):
             Arguments.
         """
         return Call(self, args, None, None)
+
+
+@tvm._ffi.register_func("relay.FunctionWithFields")
+def FunctionWithFields(function, params, body, ret_type, ty_params, attrs, virtual_device, span):
```

Comment:
I think defaulting all params after function to None will mimic the c++ version and support the above suggestion.